### ISSUE TITLE
[BUG]: Embeddings changed after update to 0.0.95

### ISSUE THREAD

**Issue by bennetrr** [serengil/deepface#1520]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [x] I have updated to the latest version of the packages.
- [x] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

0.0.95

### Python version

3.13.7

### Operating System

Windows 11, macOS 26

### Dependencies

absl-py==2.3.1
annotated-types==0.7.0
astunparse==1.6.3
azure-core==1.35.1
azure-cosmos==4.9.0
azure-identity==1.25.0
azure-keyvault-secrets==4.10.0
azure-servicebus==7.14.2
azure-storage-blob==12.26.0
beautifulsoup4==4.14.2
certifi==2025.10.5
cffi==2.0.0
charset-normalizer==3.4.3
click==8.3.0
cryptography==46.0.2
deepface==0.0.95
filelock==3.19.1
fire==0.7.1
flask==2.0.2
flask-cors==6.0.1
flatbuffers==25.9.23
gast==0.6.0
gdown==5.2.0
google-pasta==0.2.0
grpcio==1.75.1
gunicorn==23.0.0
h5py==3.14.0
idna==3.10
iniconfig==2.1.0
isodate==0.7.2
itsdangerous==2.2.0
jinja2==3.1.6
joblib==1.5.2
keras==3.11.3
libclang==18.1.1
lz4==4.4.4
markdown==3.9
markdown-it-py==4.0.0
markupsafe==3.0.3
mdurl==0.1.2
ml-dtypes==0.5.3
msal==1.34.0
msal-extensions==1.3.1
mtcnn==1.0.0
mypy==1.18.2
mypy-extensions==1.1.0
namex==0.1.0
numpy==2.2.6
opencv-python==4.12.0.88
opencv-python-headless==4.12.0.88
opt-einsum==3.4.0
optree==0.17.0
packaging==25.0
pandas==2.3.3
pathspec==0.12.1
pillow==11.3.0
pluggy==1.6.0
protobuf==6.32.1
pycparser==2.23
pydantic==2.11.10
pydantic-core==2.33.2
pygments==2.19.2
pyjwt==2.10.1
pysocks==1.7.1
pytest==8.4.2
python-dateutil==2.9.0.post0
pytz==2025.2
pyyaml==6.0.3
requests==2.32.5
retina-face==0.0.17
rich==14.1.0
ruff==0.13.3
setuptools==80.9.0
six==1.17.0
soupsieve==2.8
syrupy==5.0.0
tensorboard==2.20.0
tensorboard-data-server==0.7.2
tensorflow==2.20.0
termcolor==3.1.0
tf-keras==2.20.1
tqdm==4.67.1
types-pyyaml==6.0.12.20250915
typing-extensions==4.15.0
typing-inspection==0.4.2
tzdata==2025.2
urllib3==2.5.0
werkzeug==2.0.2
wheel==0.45.1
wrapt==1.17.3

### Reproducible example

```Python
# test_example.py
import cv2
from deepface import DeepFace

IMG_PATH = 'Mark_Zuckerberg_F8_2019_Keynote_(32830578717)_(cropped)_v_1.jpg'

def test_detect_faces(snapshot):
    img = cv2.imread(IMG_PATH)
    res = DeepFace.represent(
        img_path=img,
        detector_backend='retinaface',
        model_name='Facenet512',
    )
    assert res == snapshot

# Install version 0.0.93 of deepface and run `pytest --snapshot-update`
# Then, install the latest version of deepface and run `pytest`
```

### Expected Result

The test should pass because neither model was changed in the updates, at least according to the changelog.

### What happened instead?

The test fails because the two versions of DeepFace produce different embeddings.

### Additional Info

The following image was used for the reproducible example (we tested a few more images too, all had the same issue):

![Image](https://github.com/user-attachments/assets/f1df4168-10ab-4088-a92d-dd1e70edd4a4)

**Comment by serengil** [comment]
In 94th and 95th versions, many things changed in the preprocessing side. This will impact the embedding. Even though, you will not get the identical embedding, verification functionality will behave same. We confirmed that in unit tests already. So, this is not an issue.
